paddle/fluid/framework/ir/mkldnn/operator_scale_onednn_fuse_pass.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/operator_scale_onednn_fuse_pass.h"
+
+#include "paddle/fluid/framework/op_version_registry.h"
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+#include "paddle/fluid/string/pretty_log.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+using string::PrettyLogDetail;
+
+void FuseOperatorScaleOneDNNPass::ApplyImpl(Graph *graph) const {
+  const std::vector<std::string> fusable_ops{"fc", "matmul", "matmul_v2"};
+  for (auto op : fusable_ops) FuseScale(graph, op);
+}
+
+void FuseOperatorScaleOneDNNPass::FuseScale(Graph *graph,
+                                            const std::string &op_type) const {
+  PADDLE_ENFORCE_NOT_NULL(
+      graph, platform::errors::InvalidArgument("Graph cannot be nullptr."));
+  FusePassBase::Init(op_type + "_scale_onednn_fuse_pass", graph);
+
+  GraphPatternDetector gpd;
+  patterns::OperatorActivation op_scale_pattern(
+      gpd.mutable_pattern(), op_type + "_scale_onednn_fuse_pass");
+  op_scale_pattern(op_type, "scale");
+
+  int found_operator_scale_count = 0;
+
+  auto handler = [&](const GraphPatternDetector::subgraph_t &subgraph,
+                     Graph *g) {
+    GET_IR_NODE_FROM_SUBGRAPH(operator_op, preceding_op, op_scale_pattern);
```

Comment:
Since you are an expert in passes, what is the difference between these two first arguments(`operator_op` and `preceding_op`)? I think that in most of the passes they are equal, but in your case they are different.

Reply:
First argument is name given/used in fuse_pass.cc, second argument is name from graph_pattern_detector

Reply:
sounds like variables have bad names.

Reply:
So for example if we have generic `OperatorActivation` graph pattern detector with very general names, we can use specific node name inside fuse pass. Like instead of `preceding_op` we can use `fc_op`.

Reply:
> sounds like variables have bad names.

Yes I agree. Moreover when we set attributes, we need to get pointer to node->Op() which currently leads to `operator_op->Op()`. If you have any propositions I'm happy to change naming.

Reply:
> First argument is name given/used in `_fuse_pass.cc`, second argument is name from `graph_pattern_detector.h`

this comment suggests names like `pass_name` and `pattern_name`

> So for example if we have generic `OperatorActivation` graph pattern detector with very general names, we can use specific node name inside fuse pass. Like instead of `preceding_op` we can use `fc_op`.

but this one suggests that operator_op and preceding_op are fine.

I'm a little bit confused now.